### PR TITLE
feat!: provider enhancements, auth TUI, build ergonomics, and dry-run improvements

### LINKED ISSUE #165: feat(auth): Consistent device code and browser auth TUI experience in scafctl auth login
## Problem

The kvx TUI provides a rich `DisplaySchema` status screen for device code authentication flows (copy code, open URL actions, waiting spinner, success/error feedback). However, this experience is **not consistently applied** across all authentication handlers and flow types.

#### 1. TUI Only Triggers for Explicit `FlowDeviceCode`

In `executeLogin` (`pkg/cmd/scafctl/auth/login.go`), the kvx status TUI is only used when:

```go
if flow == auth.FlowDeviceCode && skvx.IsTerminal(ioStreams.Out) {
    return executeLoginWithStatusTUI(...)
}
```

This means the TUI is **skipped** for:
- **GitHub `FlowInteractive`** (the default) — internally runs `interactiveDeviceCodeLogin()` which uses device code + browser auto-open, but the CLI falls through to the plain-text path since `flow != FlowDeviceCode`
- Any handler that wraps device code inside another flow identifier

#### 2. GitHub Interactive Default Uses Device Code Without TUI

When a user runs `scafctl auth login github` (no explicit `--flow`):
1. Flow auto-detects to `FlowInteractive`
2. `executeLogin` sees `FlowInteractive`, **skips** TUI
3. GitHub handler routes `FlowInteractive` → `interactiveDeviceCodeLogin()` (no client_secret)
4. Handler opens browser and invokes `DeviceCodeCallback`
5. Callback was set up for **plain-text** output, not the TUI

The user gets plain-text "Waiting for authentication..." instead of the rich status screen.

#### 3. Browser Auth Code Flows Have No Waiting UX

When Entra or GitHub use browser auth code + PKCE flow (`FlowInteractive` with `client_secret`), the CLI blocks while the local HTTP server waits for the callback. The user sees no status or feedback — just a hanging terminal. The kvx status TUI could show "Waiting for browser authentication..." with a re-open URL action.

#### 4. GCP Has No TUI Feedback

GCP uses browser-based OAuth (`adcLogin`) with no device code equivalent. The browser auth code flow has no waiting UX either.

### 1. Detect Device Code Flows Generically

Instead of checking `flow == auth.FlowDeviceCode`, activate the TUI whenever a `DeviceCodeCallback` fires:
- Always set up TUI plumbing in `executeLogin`
- Launch TUI lazily when `DeviceCodeCallback` fires
- Fall back to plain text if no callback fires

### 2. Apply TUI Status Screen to All Waiting Flows

| Handler | Flow | Current UX | Proposed UX |
|---------|------|-----------|-------------|
| Entra | `device_code` | kvx status TUI | kvx status TUI (unchanged) |
| Entra | `interactive` (auth code) | Opens browser, hangs silently | Status TUI: "Waiting for browser…" with URL |
| GitHub | `interactive` (device code internally) | Plain text + browser open | kvx status TUI with copy/open actions |
| GitHub | `device_code` (explicit) | kvx status TUI | kvx status TUI (unchanged) |
| GitHub | `interactive` (auth code with secret) | Opens browser, hangs silently | Status TUI: "Waiting for browser…" |
| GCP | `interactive` (OAuth) | Opens browser, hangs silently | Status TUI: "Waiting for browser…" |

### 3. Add `BrowserAuthCallback` to `LoginOptions`

```go
type LoginOptions struct {
    // ... existing fields ...
    DeviceCodeCallback  func(userCode, verificationURI, message string)
    BrowserAuthCallback func(authURL string) // NEW: for auth code flows
}
```

The TUI shows different content based on which callback fires:
- **Device code**: code + verification URL + copy/open actions
- **Browser auth code**: "Browser opened" message + re-open URL action

## Files Involved

- `pkg/cmd/scafctl/auth/login.go` — TUI routing logic
- `pkg/auth/handler.go` — `LoginOptions`, flow constants
- `pkg/auth/github/handler.go` — GitHub flow routing
- `pkg/auth/entra/authcode_flow.go` — Entra browser auth code flow
- `pkg/auth/github/authcode_flow.go` — GitHub browser auth code flow
- `pkg/auth/gcp/handler.go` — GCP auth flows

### LINKED ISSUE #168: feat(spec): Condition (when) shorthand syntax and better error messages
## Problem

When a user puts a non-boolean, non-object value directly into a `when` clause, the error message is unhelpful:

```
failed to load solution from './solution.yaml': yaml: unmarshal errors:
  line 124: cannot unmarshal !!str `hi` into spec.Condition
```

This gives no guidance on valid forms.

## Current Behavior

`Condition` (`pkg/spec/condition.go`) only has a single `Expr` field and no custom `UnmarshalYAML`:

```go
type Condition struct {
    Expr *celexp.Expression `json:"expr" yaml:"expr" ...`
}
```

Only the explicit object form works:

```yaml
when:
  expr: "_.environment == 'prod'"
```

Any other form (boolean, string, number) produces a raw YAML unmarshal error.

### 1. Support Boolean Literals

```yaml
when: true    # always execute
when: false   # never execute (useful for temporarily disabling)
```

Stored internally as CEL expressions `true` / `false`.

### 2. Support String Shorthand for CEL Expressions

Since `Condition` only supports `expr` (not `rslvr` or `tmpl`), a string value is unambiguous:

```yaml
when: "_.environment == 'prod'"
```

Equivalent to the explicit form and far more ergonomic for the most common case.

### 3. Clear Error Messages for Invalid Types

Numbers, sequences, empty strings, and other unsupported types should produce clear, actionable errors explaining valid forms.

## Summary of Supported Forms

| Form | Example | Meaning |
|------|---------|---------|
| Boolean literal | `when: true` | Static true/false (stored as CEL `true`/`false`) |
| String shorthand | `when: "_.env == 'prod'"` | CEL expression |
| Explicit object | `when: { expr: "_.env == 'prod'" }` | CEL expression (current form) |

## Usage Across Codebase

`Condition` is used in multiple spec types:
- `Action.When` in `pkg/action/types.go`
- `Resolver.When` in `pkg/resolver/resolver.go`
- `ResolvePhase.When`, `TransformPhase.When`, `ValidatePhase.When`
- `ProviderSource.When`, `ProviderTransform.When`
- `ResolvePhase.Until` (stop condition)

## Implementation Notes

- Follow the existing `ValueRef.UnmarshalYAML` pattern in `pkg/spec/valueref.go` as a template
- Add `UnmarshalYAML` to `Condition` using `*yaml.Node` to inspect node kind/tag
- Add `UnmarshalJSON` for parity
- Boolean `!!bool` -> store as CEL expression `"true"` / `"false"`
- String `!!str` -> store as CEL expression directly (reject empty strings)
- Other scalar tags (`!!int`, `!!float`, `!!null`) -> descriptive error
- Sequence/alias nodes -> descriptive error
- Mapping node -> decode as current `{ expr: "..." }` form
- Add benchmark tests for all unmarshal paths

## Design Considerations

- String shorthand is unambiguous since `Condition` only has one field (`expr`)
- `tmpl` / `rslvr` are intentionally excluded: `Condition` is boolean-only, CEL expressions return typed values and can be validated at parse time
- Consistent with how GitHub Actions and Ansible accept expression strings directly in `when`/`if`

## Files Involved

- `pkg/spec/condition.go` - Condition type (add UnmarshalYAML/JSON)
- `pkg/spec/condition_test.go` - Add unmarshal tests
- `pkg/spec/valueref.go` - Existing shorthand pattern to follow

### LINKED ISSUE #169: feat(build): build solution ergonomic improvements (-f flag, auto-detect, version defaults)
## Summary

Improve `scafctl build solution` to be more consistent with other scafctl commands and reduce friction for common workflows.

### 1. No `-f` flag support

`run solution` and `lint` both accept `-f` to specify a solution file, but `build solution` only accepts a positional argument (`cobra.ExactArgs(1)` in `pkg/cmd/scafctl/build/solution.go`):

```sh
scafctl run solution -f ./solution.yaml        # works
scafctl lint -f ./solution.yaml                # works
scafctl build solution -f ./solution.yaml      # errors
scafctl build solution ./solution.yaml         # positional only
```

**Expected:** `-f` should work, identical to positional.

### 2. No auto-detection of solution.yaml

`run solution` and `lint` auto-detect solution files in CWD via `getter.Get()` (searches `./scafctl/solution.yaml`, `./.scafctl/solution.yaml`, `./solution.yaml`). `build solution` requires an explicit path:

```sh
scafctl lint                    # finds solution.yaml
scafctl run solution            # finds solution.yaml
scafctl build solution          # requires explicit path
```

**Expected:** `build solution` with no path should auto-detect like other commands.

### 3. `--version` should default to `metadata.version`

Currently `ResolveArtifactVersion()` in `pkg/solution/build.go` prioritizes `--version` flag over `metadata.version`, but errors if neither is provided. If the YAML declares a version, the build should use it by default:

```yaml
metadata:
  name: starter-kit
  version: 1.2.0       # use this if --version not passed
```

```sh
scafctl build solution ./solution.yaml                    # should use 1.2.0
scafctl build solution ./solution.yaml --version 2.0.0    # override to 2.0.0
```

### 4. `metadata.version` should be optional for local development

Solution validation in `pkg/solution/solution.go` enforces `metadata.version` is required for all paths, including `-f` runs. During local development this adds friction:

**Expected:**
- Running from file (`-f`): `metadata.version` optional, default to `0.0.0-dev`
- Building for catalog: `metadata.version` required (from YAML or `--version`)

### 5. Auto-increment version when omitted

When building without `--version` and no `metadata.version`, default to `0.0.1` and auto-increment on subsequent builds:

```sh
scafctl build solution ./solution.yaml         # 0.0.1
scafctl build solution ./solution.yaml         # 0.0.2
scafctl build solution ./solution.yaml --version 1.0.0  # 1.0.0 (release)
```

## Proposed CLI Behavior

| Scenario | Current | Proposed |
|---|---|---|
| `build solution -f ./solution.yaml` | Error | Works (same as positional) |
| `build solution` (no path) | Error | Auto-detect solution.yaml in CWD |
| `build` with `metadata.version: 1.2.0`, no `--version` | Error if no `--version` | Uses 1.2.0 from metadata |
| `build` with no version anywhere | Error | Uses 0.0.1, auto-increments |
| `run -f` with no `metadata.version` | Error (version required) | Works, version optional for local runs |

## Breaking Changes

None. All changes are additive.

## Files Involved

- `pkg/cmd/scafctl/build/solution.go` - Add `-f` flag, auto-detection
- `pkg/solution/build.go` - `ResolveArtifactVersion()` defaults and auto-increment
- `pkg/solution/solution.go` - Make `metadata.version` optional for file-based runs
- `pkg/solution/get/get.go` - Existing auto-discovery logic to reuse

### LINKED ISSUE #171: feat(provider/directory): Add filesOnly option to exclude directory entries from list output
## Summary

Add a `filesOnly: true` input to the directory provider to exclude directory entries from the output when only file entries are needed.

## Problem

When using the directory provider with `recursive: true` and `includeContent: true`, the output includes both file entries (with `content`) and directory entries (without `content`). This causes downstream failures when piping into `file write-tree`, which requires every entry to have a `content` property.

**Error:**

```
action execution failed: phase execution failed: action "write-static": action failed after 1 attempt(s):
input validation failed: inputs: validating root: validating /properties/entries:
validating /properties/entries/items: required: missing properties: ["content"]
```

**Current workaround** - filter entries with a CEL expression:

```yaml
entries:
  expr: '_.staticFiles.entries.filter(e, has(e.content))'
```

This works but is non-obvious boilerplate that every solution author must discover.

## Validated Behavior

- **Directory provider** (`pkg/provider/builtin/directoryprovider/directory.go`): `walkDirectory` emits ALL entries (dirs and files). No `filesOnly` option exists. Directory entries have `type: "dir"` but no `content` field.
- **File provider write-tree** (`pkg/provider/builtin/fileprovider/file.go`): Explicitly requires `content` on every entry - hard error if missing.
- **Go-template render-tree** (`pkg/provider/builtin/gotmplprovider/gotmpl.go`): Schema claims `content` is required, but code actually skips entries without content with a warning. Does not error - but silently drops directories.

## Proposed Solution

Add a `filesOnly` boolean input (default `false`) to the directory provider:

```yaml
name: read-static-files
provider: directory
inputs:
  operation: list
  path: ./static
  recursive: true
  includeContent: true
  filesOnly: true       # new: exclude directory entries from output
```

When `filesOnly: true`, directory entries are excluded from the `entries` array. Only file entries are returned.

## Implementation

Minimal, well-scoped change:

1. Add `filesOnly` to the schema (~line 80 in directory.go)
2. Add `filesOnly` field to `listOptions` struct (~line 278)
3. Parse the input in `parseListOptions` (~line 289)
4. Filter in `walkDirectory` (~line 490): skip emitting if `isDir && opts.filesOnly`

## Use Case

The `directory -> go-template render-tree -> file write-tree` pipeline is a documented first-class pattern. The most common use of `directory list` with `includeContent: true` is to feed entries into `render-tree` or `write-tree`, both of which expect file entries with content. A `filesOnly` option makes this pipeline work without defensive CEL filters.

## Impact

- **Low risk** - additive, opt-in with `false` default; no breaking changes
- **High value** - eliminates a common pitfall in the standard template rendering pipeline

## Files Involved

- `pkg/provider/builtin/directoryprovider/directory.go` - Add `filesOnly` input, schema, parsing, and filtering

### LINKED ISSUE #258: feat(exec): add passthrough option for streaming command output to terminal
# feat(exec): add `passthrough` option for streaming command output to terminal

## Summary

The exec provider always captures stdout/stderr into its result object. There is no way to stream command output directly to the user's terminal. This makes scafctl awkward as a task runner -- commands with formatted output (colored logs, progress bars) lose all formatting when captured and replayed.

## Problem

~~~
# Running lint directly -- nice formatted output
scafctl lint solution -f ./solution.yaml

# Running lint via taskfile -- raw captured output
scafctl run solution -f taskfile.yaml -r task=lint
# Output: raw blob, no formatting, no colors
~~~

## Proposal

Add a `passthrough` input to the exec provider:

~~~yaml
lint:
  provider: exec
  inputs:
    command: scafctl
    args: ["lint", "solution", "-f", "./solution.yaml"]
    passthrough: true
~~~

### Behavior when `passthrough: true`

- stdout/stderr stream directly to the user's terminal in real-time
- Colors, formatting, progress indicators, and TTY features are preserved
- Exit code is still captured for action success/failure determination
- The result `stdout`/`stderr` fields are empty (output was streamed, not captured)

### Behavior when `passthrough: false` (default)

- No change from current behavior

## Precedent

| Tool | Default | Control |
|------|---------|---------|
| Task (taskfile.dev) | Streams to terminal | `silent: true` to suppress |
| Make | Streams to terminal | `@` prefix to suppress |
| scafctl exec | Captures (silent) | **No way to stream** |

## Considerations

- **Parallel actions**: passthrough from concurrent actions could produce interleaved output. Consider buffering when `max-action-concurrency > 1`.
- **`__actions` references**: downstream actions depending on `__actions["name"].results.stdout` get empty strings with passthrough.
- **Testing**: functional tests cannot assert on passthrough output. Use `passthrough: false` in tests.
- **TTY detection**: ideally the child process should inherit the parent's TTY.

## Scope

**S-M** -- Core change is wiring stdout/stderr to the parent terminal. Schema update, docs, and edge case handling for parallel actions.

## Relationship to Other Issues

- `feat-exec-output-shorthand` -- the `output` input controls what the provider *returns*; `passthrough` controls whether output is *streamed*. They are complementary.
- `feat-action-filter` -- action filtering + passthrough together make scafctl a proper task runner.

### PR DESCRIPTION
**Pull request by abaker-9**
- add move/rename operation to directory provider with cross-device fallback (copy + remove)
- add list_commit_pulls operation to GitHub provider via REST API
- add passthrough mode to exec provider for streaming terminal output
- write-tree now silently skips directory entries without content, removing the need for CEL .filter(e, has(e.content)) workarounds
- add actions.yaml and actions.yml to solution auto-discovery filenames
- display solution name, version, and source path after loading
- sort dry-run action plan by section (actions before finally) and phase
- add consistent device code and browser auth TUI experience across all auth handlers including GCP ADC flow
- add condition shorthand syntax with expr/expression field support and improved error messages
- add auto-increment patch version and name mismatch validation for build solution command
- add kvx status screen TUI for browser auth waiting state
- clean up duplicate .gitignore entry and fix go.mod dependency

BREAKING CHANGE: write-tree no longer errors on entries missing content -- they are silently skipped as directory entries

Closes #168
Closes #258
Closes #165
Closes #169
Closes #171

### CODE REVIEW

File: pkg/spec/condition.go [hunk 1]
```diff
@@ -5,17 +5,169 @@ package spec
...
+		return fmt.Errorf("invalid condition at line %d, column %d: %w", node.Line, node.Column, err)
+	}
+	if raw.Expr != nil && raw.Expression != nil {
+		return fmt.Errorf("invalid condition at line %d, column %d: specify either 'expr' or 'expression', not both", node.Line, node.Column)
+	}
+	if raw.Expression != nil {
+		c.Expr = raw.Expression
+	} else {
+		c.Expr = raw.Expr
+	}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/spec/condition.go:87]: Fixed — added empty-string validation and missing-key validation in `unmarshalMappingYAML`. Now rejects `{expr: ""}`, `{expression: ""}`, and `{foo: bar}` with actionable error messages. Tests added.

File: pkg/spec/condition.go [hunk 2]
```diff
@@ -5,17 +5,169 @@ package spec
...
+		return fmt.Errorf("invalid condition: expected boolean, string, or object {\"expr\": \"...\"}: %w", err)
+	}
+	if obj.Expr != nil && obj.Expression != nil {
+		return fmt.Errorf("invalid condition: specify either 'expr' or 'expression', not both")
+	}
+	if obj.Expression != nil {
+		c.Expr = obj.Expression
+	} else {
+		c.Expr = obj.Expr
+	}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/spec/condition.go:162]: Fixed — added empty-string validation in `UnmarshalJSON` object path. Tests added for both `{"expr": ""}` and `{"expression": ""}`.

File: pkg/auth/gcp/adc_flow.go [hunk 3]
```diff
@@ -75,6 +75,10 @@ func (h *Handler) adcLogin(ctx context.Context, opts auth.LoginOptions) (*auth.R
 	lgr.V(1).Info("opening browser for authentication", "url", authURL)
 	if err := oauth.OpenBrowser(ctx, authURL); err != nil {
 		lgr.V(0).Info("failed to open browser, please open this URL manually", "url", authURL)
+		// Notify callback so the CLI can display the URL to the user.
+		if opts.DeviceCodeCallback != nil {
+			opts.DeviceCodeCallback("", authURL, "Open this URL in your browser to authenticate")
+		}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/auth/gcp/adc_flow.go]: Added clarifying comment explaining the reuse of DeviceCodeCallback with empty userCode for browser-based ADC auth. A dedicated BrowserAuthCallback is a good future enhancement but would require changes across all auth handlers — keeping it as-is for now with documentation.

File: pkg/terminal/kvx/output.go [hunk 4]
```diff
@@ -554,11 +568,21 @@ func (o *OutputOptions) writeStructured(data any) error {
 		}
 	}
 
+	// Print scalar values as plain text instead of JSON wrapping.
+	if isScalar(outputData) {
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/terminal/kvx/output.go]: Good catch, fixed. Scalar fast-path now only applies for non-structured formats using `IsStructuredFormat()`. JSON/YAML/CSV/TOML go through their proper serializers.

File: pkg/terminal/kvx/output.go [hunk 5]
```diff
@@ -719,3 +743,141 @@ func StructToMap(v any) (any, error) {
...
+		for i, m := range v {
+			items[i] = m
+		}
+		return o.writeCSVSlice(w, items)
+	case map[string]any:
+		return o.writeCSVSlice(w, []any{v})
+	default:
+		// Scalar or unsupported type -- write as single value
+		return w.Write([]string{fmt.Sprintf("%v", data)})
+	}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/terminal/kvx/output.go]: Good catch, fixed. Removed `defer w.Flush()`, now flush explicitly and check `w.Error()` before returning.

File: pkg/provider/builtin/directoryprovider/directory.go [hunk 6]
```diff
@@ -750,6 +775,52 @@ func (p *DirectoryProvider) executeCopy(ctx context.Context, absPath string, inp
...
+	// Try atomic rename first (works on same device).
+	if renameErr := os.Rename(absPath, absDest); renameErr != nil {
+		// Fall back to copy + remove for cross-device moves.
+		if err := copyDir(absPath, absDest); err != nil {
+			return nil, fmt.Errorf("failed to move directory (copy phase): %w", err)
+		}
+		if err := os.RemoveAll(absPath); err != nil {
+			return nil, fmt.Errorf("failed to move directory (remove source phase): %w", err)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/provider/builtin/directoryprovider/directory.go:817]: Fixed — copy+remove fallback is now gated on `errors.Is(renameErr, syscall.EXDEV)`. All other rename errors (permissions, destination exists, etc.) are returned directly.

File: pkg/provider/builtin/fileprovider/file.go [hunk 7]
```diff
@@ -840,7 +842,8 @@ func (p *FileProvider) parseWriteTreeEntries(inputs map[string]any) ([]writeTree
 
 		content, ok := entry["content"].(string)
 		if !ok {
-			return nil, fmt.Errorf("entries[%d].content is required and must be a string", i)
+			// Skip directory entries (no content field) silently.
+			continue
 		}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/provider/builtin/fileprovider/file.go]: Fixed — now only skips when `content` key is absent. If `content` is present but wrong type (e.g. `content: 123`), returns a validation error. Test added for non-string content.

File: pkg/cmd/scafctl/run/common.go [hunk 8]
```diff
@@ -597,11 +597,36 @@ func (o *sharedResolverOptions) prepareSolutionForExecution(ctx context.Context)
...
+		if w.VerboseEnabled() {
+			w.Verbosef("Solution loaded: %s (version=%s, dir=%s)",
+				name, ver, result.SolutionDir)
+		}
+
+		// Show a concise summary so the user knows what was resolved.
+		// Skip when structured output is requested to avoid polluting JSON/YAML.
+		if o.Output != "json" && o.Output != "yaml" {
+			switch {
+			case name != "" && ver != "" && source != "":
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/run/common.go:632]: Fixed — now uses `kvx.ParseOutputFormat` + `kvx.IsStructuredFormat()` to suppress for all structured formats (JSON, YAML, CSV, TOML, Mermaid) plus quiet mode.

File: pkg/provider/builtin/directoryprovider/directory.go [hunk 9]
```diff
@@ -750,6 +775,52 @@ func (p *DirectoryProvider) executeCopy(ctx context.Context, absPath string, inp
...
+		return nil, fmt.Errorf("failed to stat source: %w", err)
+	}
+	if !info.IsDir() {
+		return nil, fmt.Errorf("source is not a directory: %s", absPath)
+	}
+
+	// Try atomic rename first (works on same device).
+	if renameErr := os.Rename(absPath, absDest); renameErr != nil {
+		// Fall back to copy + remove for cross-device moves.
+		if err := copyDir(absPath, absDest); err != nil {
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/provider/builtin/directoryprovider/directory.go:811]: Fixed — fallback is now gated on syscall.EXDEV only. Thanks!

File: pkg/provider/builtin/fileprovider/file.go [hunk 10]
```diff
@@ -840,7 +842,8 @@ func (p *FileProvider) parseWriteTreeEntries(inputs map[string]any) ([]writeTree
 
 		content, ok := entry["content"].(string)
 		if !ok {
-			return nil, fmt.Errorf("entries[%d].content is required and must be a string", i)
+			// Skip directory entries (no content field) silently.
+			continue
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/provider/builtin/fileprovider/file.go]: Fixed — now only skips entries where content key is absent. Non-string content returns a validation error.

File: pkg/cmd/scafctl/run/common.go [hunk 11]
```diff
@@ -597,11 +597,36 @@ func (o *sharedResolverOptions) prepareSolutionForExecution(ctx context.Context)
...
+
+		if w.VerboseEnabled() {
+			w.Verbosef("Solution loaded: %s (version=%s, dir=%s)",
+				name, ver, result.SolutionDir)
+		}
+
+		// Show a concise summary so the user knows what was resolved.
+		// Skip when structured output is requested to avoid polluting JSON/YAML.
+		if o.Output != "json" && o.Output != "yaml" {
+			switch {
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/cmd/scafctl/run/common.go]: Fixed — uses kvx.IsStructuredFormat() to suppress for all structured outputs.

File: pkg/config/types.go [hunk 12]
```diff
@@ -455,6 +456,13 @@ func (b *BuildConfig) IsAutoCacheRemoteArtifacts() bool {
 }
 
 // APIServerConfig holds REST API server configuration.
+// DiscoveryConfig holds auto-discovery preferences.
+type DiscoveryConfig struct {
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/config/types.go]: Fixed -- moved the `// APIServerConfig` comment down to its struct and gave `DiscoveryConfig` its own header comment. Good catch.

File: pkg/config/types.go [hunk 13]
```diff
@@ -455,6 +456,13 @@ func (b *BuildConfig) IsAutoCacheRemoteArtifacts() bool {
...
+	// ActionFiles overrides the file names searched during "run action"
+	// auto-discovery. When empty, the built-in defaults are used.
+	ActionFiles []string `json:"actionFiles,omitempty" yaml:"actionFiles,omitempty" mapstructure:"actionFiles" doc:"File names for action auto-discovery" maxItems:"20" example:"[\"actions.yaml\",\"actions.yml\"]"`
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/config/types.go]: Good catch, fixed. Removed the `example` tag to match the convention used by other slice fields in this file.

File: pkg/spec/condition.go [hunk 14]
```diff
@@ -5,17 +5,180 @@ package spec
...
+
+// UnmarshalJSON implements custom JSON unmarshalling for Condition.
+// It supports boolean literals, string shorthand, and the explicit {"expr": "..."} form.
+func (c *Condition) UnmarshalJSON(data []byte) error {
+	*c = Condition{}
+
+	// Null -> zero-value Condition (nil Expr).
+	if string(data) == "null" {
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/spec/condition.go:121]: Good catch, fixed. Added `bytes.TrimSpace(data)` before the null comparison.

[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/spec/condition.go:121]: Good catch, fixed. Added bytes.TrimSpace before the null comparison.
